CMakeLists.txt

```diff
@@ -1,38 +1,93 @@
+set(SVF_INSTALL_INCLUDE_DIR ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}/svf)
+
+# Set where extapi.bc is exported to in the installed CMake package
+set(SVF_INSTALL_EXTAPI_DIR ${SVF_INSTALL_INCLUDE_DIR}/SVF-LLVM)
+set(SVF_INSTALL_EXTAPI_FILE ${SVF_INSTALL_EXTAPI_DIR}/extapi.bc)
+
+message(STATUS "Building ${PROJECT_NAME} with configuration:
+  SVF version:                                  ${SVF_VERSION}
+  SVF root directory:                           ${SVF_SOURCE_DIR}
+  SVF binary directory:                         ${SVF_BINARY_DIR}
+  SVF option - create sanitiser build (str):    ${SVF_SANITIZE}
+  SVF option - coverage build:                  ${SVF_COVERAGE}
+  SVF option - warnings as errors:              ${SVF_WARN_AS_ERROR}
+  SVF option - unused dynamic symbols:          ${SVF_EXPORT_DYNAMIC}
+  SVF option - enable build assertions:         ${SVF_ENABLE_ASSERTIONS}")
+
+message(STATUS "Using CMake build configuration:
+  CMake generator:                              ${CMAKE_GENERATOR}
+  CMake C compiler:                             ${CMAKE_C_COMPILER_ID}
+  CMake C++ compiler:                           ${CMAKE_CXX_COMPILER_ID}
+  CMake current list directory:                 ${CMAKE_CURRENT_LIST_DIR}
+  CMake current source directory:               ${CMAKE_CURRENT_SOURCE_DIR}
+  CMake current binary directory:               ${CMAKE_CURRENT_BINARY_DIR}")
+
+# Create config.h based on config.in
+configure_file(${SVF_SOURCE_DIR}/.config.in ${SVF_BINARY_DIR}/include/Util/config.h)
+
+# Include the directory where the configuration header is exported for all targets
+include_directories(${SVF_BINARY_DIR}/include)
 
-# Treat compiler warnings as errors
-add_compile_options("-Werror" "-Wall" "-Wno-deprecated-declarations")
+# Install generated configuration header (see `configure_file()`) to top-level include dir of SVF
+install(
+  FILES ${SVF_BINARY_DIR}/include/Util/config.h
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/svf/Util
+)
 
 # Build SVF with C++ standard C++17
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_EXTENSIONS ON)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
+# By default, build SVF and its targets treating all compiler warnings as errors (except deprecations)
+add_compile_options(
+  $<$<BOOL:${SVF_WARN_AS_ERROR}>:-Wall>
+  $<$<BOOL:${SVF_WARN_AS_ERROR}>:-Werror>
+  $<$<BOOL:${SVF_WARN_AS_ERROR}>:-Wno-deprecated-declarations>
+)
+
 # Keep assertions enabled if requested
-option(SVF_ENABLE_ASSERTIONS "Always enable assertions")
-if(SVF_ENABLE_ASSERTIONS)
-  add_compile_options("-UNDEBUG")
-endif()
+add_compile_options($<$<BOOL:${SVF_ENABLE_ASSERTIONS}>:-UNDEBUG>)
 
-# Turn this on if you need symbols (e.g., use them for backtrace debugging)
-# add_link_options("-rdynamic")
+# Export dynamic symbols if requested (adds "-export-dynamic" to linkers that support it to enable backtraces)
+add_compile_options($<$<BOOL:${SVF_EXPORT_DYNAMIC}>:-rdynamic>)
 
-option(SVF_COVERAGE "Create coverage build")
```

Comment:
Coverage file looks missing: 
https://github.com/SVF-tools/SVF/actions/runs/7349649179/job/20021425836?pr=1305#step:17:19

Reply:
My bad, I think this was because I forgot to check if `SVF_COVERAGE` is defined as an environment variable (I only checked for it as a command-line argument).